src/build_system.ml

```diff
@@ -448,6 +385,12 @@ type t =
   ; hook : hook -> unit
   ; (* Package files are part of *)
     packages : Package.Name.t Path.Table.t
+  ; load_dir_cache : Path.Set.t Memoize.t
+  (* memoized functions *)
+  ; prepare_rule_def : (Internal_rule.t * Static_deps.t * (unit, Action.t) Build.t, Action.t * Deps.t) CRef.comp
```

Comment:
I've now fixed this, but it does mean that rule.static_deps (which is defined as lazy) is evaluated twice, once in `prepare_rule` and once in `build_rule`. Otherwise one would need to memoize this.

Reply:
Hmm, evaluating it twice doesn't seem good. We should keep it separate then

Reply:
I've decided to introduce a `Once` module, which similarly to lazy only evaluates something when `Once.get` is called, but that also caches the result so `Once.get` only causes the expression to be evaluated a single time.